onnx/src/data_resolver.rs

```diff
@@ -0,0 +1,81 @@
+use std::fs::File;
+use std::io::{BufRead, BufReader};
+use std::path::Path;
+use tract_hir::internal::*;
+
+use tract_hir::internal::TractResult;
+
+#[cfg(not(target_family = "wasm"))]
+pub fn default() -> Box<dyn ModelDataResolver> {
+    Box::new(MmapDataResolver)
+}
+
+#[cfg(target_family = "wasm")]
+pub fn default() -> Box<dyn ModelDataResolver> {
+    Box::new(FopenDataResolver)
+}
+
+pub trait ModelDataResolver {
```

Comment:
This should work @kali 

Not too important but I wonder if these methods can be consolidated to make the trait simpler.

Reply:
I'm not sure we can without having it lead to extra copies here and there. Can you see a way ?

Reply:
Hmm where is `read()` used? Couldn't we just have `read_bytes_from_path`?

Reply:
indeed. not sure why I incuded it in the first place.